[BUG] cudf_udf failed in all spark release intermittently

**Describe the bug**
rapids_databricks_nightly-dev, ID 19, 20
rapids_integration-dev spark-301 302, ID 186, 187
rapids_it-3.1.x-SNAPSHOT-dev spark-312-SNAPSHOT, ID 149, 150

this actually is not 100% reproducible, and it could fail in all envs
cudf_udf integration tests failed,
```
[2021-05-27T08:17:32.021Z] Traceback (most recent call last):
[2021-05-27T08:17:32.021Z]   File "/home/ubuntu/spark-rapids/dist/target/rapids-4-spark_2.12-21.06.0-SNAPSHOT.jar/rapids/daemon_databricks.py", line 132, in manager
[2021-05-27T08:17:32.021Z]   File "/home/ubuntu/spark-rapids/dist/target/rapids-4-spark_2.12-21.06.0-SNAPSHOT.jar/rapids/worker.py", line 37, in initialize_gpu_mem
[2021-05-27T08:17:32.021Z]     from cudf import rmm
[2021-05-27T08:17:32.021Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/__init__.py", line 11, in <module>
[2021-05-27T08:17:32.021Z]     from cudf import core, datasets, testing
[2021-05-27T08:17:32.021Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/core/__init__.py", line 3, in <module>
[2021-05-27T08:17:32.021Z]     from cudf.core import _internals, buffer, column, column_accessor, common
[2021-05-27T08:17:32.021Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/core/_internals/__init__.py", line 3, in <module>
[2021-05-27T08:17:32.021Z]     from cudf.core._internals.where import where
[2021-05-27T08:17:32.021Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/core/_internals/where.py", line 11, in <module>
[2021-05-27T08:17:32.022Z]     from cudf.core.column import ColumnBase
[2021-05-27T08:17:32.022Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/core/column/__init__.py", line 17, in <module>
[2021-05-27T08:17:32.022Z]     from cudf.core.column.datetime import DatetimeColumn  # noqa: F401
[2021-05-27T08:17:32.022Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/core/column/datetime.py", line 20, in <module>
[2021-05-27T08:17:32.022Z]     from cudf.core.column import (
[2021-05-27T08:17:32.022Z]   File "/databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/core/column/string.py", line 78, in <module>
[2021-05-27T08:17:32.022Z]     from cudf._lib.strings.combine import (
[2021-05-27T08:17:32.022Z] ImportError: /databricks/conda/envs/databricks-ml-gpu/lib/python3.7/site-packages/cudf/_lib/strings/combine.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZN4cudf7strings18join_list_elementsERKNS_17lists_column_viewERKNS_19strings_column_viewERKNS_13string_scalarES9_NS0_18separator_on_nullsEPN3rmm2mr22device_memory_resourceE
```

```
[2021-05-27T08:18:59.417Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling o2591.collectToPython.[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 5 in stage 35.0 failed 1 times, most recent failure: Lost task 5.0 in stage 35.0 (TID 220, 10.2.128.4, executor driver): java.net.SocketException: Connection reset[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at java.net.SocketInputStream.read(SocketInputStream.java:210)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at java.net.SocketInputStream.read(SocketInputStream.java:141)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at java.net.SocketInputStream.read(SocketInputStream.java:224)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at java.io.DataInputStream.readInt(DataInputStream.java:387)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.createSocket$1(PythonWorkerFactory.scala:210)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.liftedTree1$1(PythonWorkerFactory.scala:233)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.createThroughDaemon(PythonWorkerFactory.scala:225)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.create(PythonWorkerFactory.scala:119)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.SparkEnv.createPythonWorker(SparkEnv.scala:192)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.api.python.BasePythonRunner.compute(PythonRunner.scala:183)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.sql.execution.python.PandasGroupUtils$.executePython(PandasGroupUtils.scala:44)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.sql.execution.python.rapids.GpuPandasUtils$.executePython(GpuPandasUtils.scala:35)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.sql.rapids.execution.python.GpuFlatMapCoGroupsInPandasExec.$anonfun$doExecute$1(GpuFlatMapCoGroupsInPandasExec.scala:138)[0m
[2021-05-27T08:18:59.417Z] [1m[31mE                   	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:101)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:356)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:320)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:144)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:117)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$9(Executor.scala:640)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1581)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:643)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   [0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   Driver stacktrace:[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2519)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2466)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2460)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2460)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1152)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1152)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at scala.Option.foreach(Option.scala:407)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1152)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2721)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2668)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2656)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:938)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2339)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2434)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.runSparkJobs(Collector.scala:273)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.collect(Collector.scala:308)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:82)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:88)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:508)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:480)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeCollectResult(SparkPlan.scala:401)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3497)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3709)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$5(SQLExecution.scala:116)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:249)[0m
[2021-05-27T08:18:59.418Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:101)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:845)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:77)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:199)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3707)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3495)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at sun.reflect.GeneratedMethodAccessor219.invoke(Unknown Source)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:295)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at py4j.GatewayConnection.run(GatewayConnection.java:251)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   Caused by: java.net.SocketException: Connection reset[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.net.SocketInputStream.read(SocketInputStream.java:210)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.net.SocketInputStream.read(SocketInputStream.java:141)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.net.SocketInputStream.read(SocketInputStream.java:224)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.io.DataInputStream.readInt(DataInputStream.java:387)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.createSocket$1(PythonWorkerFactory.scala:210)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.liftedTree1$1(PythonWorkerFactory.scala:233)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.createThroughDaemon(PythonWorkerFactory.scala:225)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.api.python.PythonWorkerFactory.create(PythonWorkerFactory.scala:119)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.SparkEnv.createPythonWorker(SparkEnv.scala:192)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.api.python.BasePythonRunner.compute(PythonRunner.scala:183)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.execution.python.PandasGroupUtils$.executePython(PandasGroupUtils.scala:44)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.execution.python.rapids.GpuPandasUtils$.executePython(GpuPandasUtils.scala:35)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.sql.rapids.execution.python.GpuFlatMapCoGroupsInPandasExec.$anonfun$doExecute$1(GpuFlatMapCoGroupsInPandasExec.scala:138)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:101)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:356)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:320)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:144)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:117)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$9(Executor.scala:640)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1581)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:643)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2021-05-27T08:18:59.419Z] [1m[31mE                   	... 1 more[0m
```

--- a/jenkins/databricks/test.sh
+++ b/jenkins/databricks/test.sh
@@ -59,15 +59,17 @@ if [ -d "$LOCAL_JAR_PATH" ]; then
     LOCAL_JAR_PATH=$LOCAL_JAR_PATH bash $LOCAL_JAR_PATH/integration_tests/run_pyspark_from_build.sh  --runtime_env="databricks" --test_type=$TEST_TYPE
 
     ## Run cudf-udf tests
-    CUDF_UDF_TEST_ARGS="$CUDF_UDF_TEST_ARGS --conf spark.executorEnv.PYTHONPATH=`ls $LOCAL_JAR_PATH/rapids-4-spark_*.jar | grep -v 'tests.jar'`"
-    LOCAL_JAR_PATH=$LOCAL_JAR_PATH SPARK_SUBMIT_FLAGS="$SPARK_CONF $CUDF_UDF_TEST_ARGS" TEST_PARALLEL=1 \
-        bash $LOCAL_JAR_PATH/integration_tests/run_pyspark_from_build.sh --runtime_env="databricks" -m "cudf_udf" --cudf_udf --test_type=$TEST_TYPE
+    #disable cudf_udf tests for now until we resolve https://github.com/NVIDIA/spark-rapids/issues/2521
+    #CUDF_UDF_TEST_ARGS="$CUDF_UDF_TEST_ARGS --conf spark.executorEnv.PYTHONPATH=`ls $LOCAL_JAR_PATH/rapids-4-spark_*.jar | grep -v 'tests.jar'`"
+    #LOCAL_JAR_PATH=$LOCAL_JAR_PATH SPARK_SUBMIT_FLAGS="$SPARK_CONF $CUDF_UDF_TEST_ARGS" TEST_PARALLEL=1 \
+    #    bash $LOCAL_JAR_PATH/integration_tests/run_pyspark_from_build.sh --runtime_env="databricks" -m "cudf_udf" --cudf_udf --test_type=$TEST_TYPE
 else
     ## Run tests with jars building from the spark-rapids source code
     bash /home/ubuntu/spark-rapids/integration_tests/run_pyspark_from_build.sh --runtime_env="databricks" --test_type=$TEST_TYPE
 
     ## Run cudf-udf tests
-    CUDF_UDF_TEST_ARGS="$CUDF_UDF_TEST_ARGS --conf spark.executorEnv.PYTHONPATH=`ls /home/ubuntu/spark-rapids/dist/target/rapids-4-spark_*.jar | grep -v 'tests.jar'`"
-    SPARK_SUBMIT_FLAGS="$SPARK_CONF $CUDF_UDF_TEST_ARGS" TEST_PARALLEL=1 \
-        bash /home/ubuntu/spark-rapids/integration_tests/run_pyspark_from_build.sh --runtime_env="databricks"  -m "cudf_udf" --cudf_udf --test_type=$TEST_TYPE
+    #disable cudf_udf tests for now until we resolve https://github.com/NVIDIA/spark-rapids/issues/2521
+    #CUDF_UDF_TEST_ARGS="$CUDF_UDF_TEST_ARGS --conf spark.executorEnv.PYTHONPATH=`ls /home/ubuntu/spark-rapids/dist/target/rapids-4-spark_*.jar | grep -v 'tests.jar'`"
+    #SPARK_SUBMIT_FLAGS="$SPARK_CONF $CUDF_UDF_TEST_ARGS" TEST_PARALLEL=1 \
+    #    bash /home/ubuntu/spark-rapids/integration_tests/run_pyspark_from_build.sh --runtime_env="databricks"  -m "cudf_udf" --cudf_udf --test_type=$TEST_TYPE
 fi

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -92,7 +92,10 @@ echo "----------------------------START TEST------------------------------------
 pushd $RAPIDS_INT_TESTS_HOME
 TEST_TYPE="nightly"
 spark-submit $BASE_SPARK_SUBMIT_ARGS --jars $RAPIDS_TEST_JAR ./runtests.py -v -rfExXs --std_input_path="$WORKSPACE/integration_tests/src/test/resources/" --test_type=$TEST_TYPE
-spark-submit $BASE_SPARK_SUBMIT_ARGS $CUDF_UDF_TEST_ARGS --jars $RAPIDS_TEST_JAR ./runtests.py -m "cudf_udf" -v -rfExXs --cudf_udf --test_type=$TEST_TYPE
+
+#disable cudf_udf tests for now until we resolve https://github.com/NVIDIA/spark-rapids/issues/2521
+#spark-submit $BASE_SPARK_SUBMIT_ARGS $CUDF_UDF_TEST_ARGS --jars $RAPIDS_TEST_JAR ./runtests.py -m "cudf_udf" -v -rfExXs --cudf_udf --test_type=$TEST_TYPE
+
 #only run cache tests with our serializer in nightly test for Spark version >= 3.1.1
 if [ "$(printf '%s\n' "3.1.1" "$SPARK_VER" | sort -V | head -n1)" = "3.1.1" ]; then
   SHIM_PACKAGE=$(echo ${SPARK_VER} | sed 's/\.//g' | sed 's/-SNAPSHOT//')

